### ISSUE TITLE
APIs to access the classes.dex

### ISSUE THREAD

**Issue by duanjason** [JesusFreke/smali#661]
Hello,

This is more a question than defect. Is there any documentation on this project?

Is there any APIs that I can use to access the classes.dex? For example, giving all classes in the dex, modifying a class method, such as insert some code to a method or remove some code from it etc.

**Comment by xpirt** [comment]
You can check the [wiki](https://github.com/JesusFreke/smali/wiki) for an intial insight. Also visit the official [Dalvik bytecode documentation](https://source.android.com/devices/tech/dalvik/dalvik-bytecode) for info about the `dex` format, instructions, etc.

By the way, apktool uses smali to decompress .dex files within an apk.

**Comment by Lanchon** [comment]
yes there is such an api in the dexlib2 maven artifact (part of smali). if you are concerned with possibly multidex APKs, you might want to take a look at multidexlib2.
